File: ExampleOC/ExampleNotificationService/NotificationService.m
//
//  NotificationService.m
//  ExampleNotificationService
//
//  Created by Jimmy on 2023/7/18.
//

#import "NotificationService.h"
@import CinnoxVisitorCoreSDK;

@interface NotificationService ()

@property (nonatomic, strong) id<CinnoxNotificationServiceHandler> handler;

@end

@implementation NotificationService

- (void)didReceiveNotificationRequest:(UNNotificationRequest *)request withContentHandler:(void (^)(UNNotificationContent * _Nonnull))contentHandler {
    self.handler = [CinnoxVisitorCore createNotificationServiceHandler];
    
    [self.handler didReceive: request withCinnoxContentHandler:^(UNNotificationContent * _Nonnull content) {
        contentHandler(content);
    } nonCinnoxContent:^{
        UNMutableNotificationContent *bestAttemptContent = [request.content mutableCopy];
        contentHandler(bestAttemptContent);
    }];
}

@end

File: ExampleSwift/ExampleNotificationService/NotificationService.swift
//
//  NotificationService.swift
//  ExampleNotificationService
//
//  Created by Jimmy on 2023/7/17.
//

import UserNotifications
import CinnoxVisitorCoreSDK

class NotificationService: UNNotificationServiceExtension {
    public var notificationHandler = CinnoxVisitorCore.createNotificationServiceHandler()

    override func didReceive(_ request: UNNotificationRequest, withContentHandler contentHandler: @escaping (UNNotificationContent) -> Void) {
        notificationHandler.didReceive(request) { content in
            contentHandler(content)
        } nonCinnoxContent: {
            if let bestAttemptContent = (request.content.mutableCopy() as? UNMutableNotificationContent) {
                contentHandler(bestAttemptContent)
            }
        }
    }
}

